client-lite/src/trace/do_log.h

```diff
@@ -1,24 +1,36 @@
 #pragma once
 
-void DoTraceLoggingRegister();
-void DoTraceLoggingUnregister();
+#include <string>
 
-bool IsLoggingEnabled(boost::log::trivial::severity_level level);
-bool IsVerboseLoggingEnabled();
+namespace DOLog
+{
 
-void LogRuntimeFailure(const docli::FailureInfo& failure) noexcept;
+enum class Level
+{
+    Error,
+    Warning,
+    Info,
+    Verbose,
+};
 
-#define EVENT_LEVEL_ERROR       boost::log::trivial::error
-#define EVENT_LEVEL_WARNING     boost::log::trivial::warning
-#define EVENT_LEVEL_INFO        boost::log::trivial::info
-#define EVENT_LEVEL_VERBOSE     boost::log::trivial::trace
+void Init(const std::string& logDir, Level maxLogLevel);
+void Close();
+void Write(Level level, const char* pszFunc, unsigned int uLine, const char* pszFmt, ...);
+void WriteResult(Level level, const char* pszFunc, unsigned int uLine, HRESULT hr, const char* pszFmt, ...);
+
+} // namespace DOLog
+
+#define EVENT_LEVEL_ERROR       DOLog::Level::Error
+#define EVENT_LEVEL_WARNING     DOLog::Level::Warning
+#define EVENT_LEVEL_INFO        DOLog::Level::Info
+#define EVENT_LEVEL_VERBOSE     DOLog::Level::Verbose
 
 // '##' is required before __VA_ARGS__ to allow an empty arg list to the variadic macro.
 // MSVC supports this by default but GCC requires '##'. C++2a has added VA_OPT macro
 // to officially support this behavior.
 
-#define DoLogMessage(level, msg, ...)   ::LogMessage((level), __FUNCTION__, __LINE__, (msg), ##__VA_ARGS__)
-#define DoLogResult(lvl, hr, msg, ...)  ::LogResult((lvl), __FUNCTION__, __LINE__, (hr), (msg), ##__VA_ARGS__)
+#define DoLogMessage(level, msg, ...)   DOLog::Write((level), __FUNCTION__, __LINE__, (msg), ##__VA_ARGS__)
+#define DoLogResult(lvl, hr, msg, ...)  DOLog::WriteResult((lvl), __FUNCTION__, __LINE__, (hr), (msg), ##__VA_ARGS__)
```

Comment:
>lvl [](http://example.com/codeflow?start=20&length=3)

nit: should this be level, for consistency with line 32?

Reply:
Yes, will do.